src/test/java/uk/gov/ons/ctp/response/casesvc/message/TestPubSubMessage.java

```diff
@@ -32,14 +33,36 @@ public CaseNotificationDTO getPubSubCaseNotification() throws IOException {
           consumer.ack();
         };
     Subscriber subscriber = pubsubEmulator.getEmulatorSubscriber(receiver);
-    Thread t =
-        new Thread() {
+    Thread t = new Thread(() -> subscriber.startAsync().awaitRunning());
+    ExecutorService service = Executors.newSingleThreadExecutor();
+    service.execute(t);
+    while (receivedMessage == null) {
```

Comment:
could it be an empty value

Reply:
yes it can be null. So basically its a thread based approach where in the executer keeps checking if the emulator has received the message on the subscription to be picked up.